File: C++/longest-increase-streak.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<long long> arr(n);
    for (auto &x : arr) cin >> x;

    if (n == 0) {
        cout << 0 << endl;
        return 0;
    }

    int maxLen = 1, curLen = 1;

    for (int i = 1; i < n; i++) {
        if (arr[i] > arr[i - 1]) {
            curLen++;
        } else {
            curLen = 1;
        }
        maxLen = max(maxLen, curLen);
    }

    cout << maxLen << endl;
    return 0;
}

File: JS/longest-increase-streak.js
function longestIncreasingStreak(arr) {
    if (arr.length === 0) return 0;

    let maxLen = 1;
    let curLen = 1;

    for (let i = 1; i < arr.length; i++) {
        if (arr[i] > arr[i - 1]) {
            curLen++;
        } else {
            curLen = 1;
        }
        maxLen = Math.max(maxLen, curLen);
    }

    return maxLen;
}

// --- Read input from stdin ---
const lines = require('fs').readFileSync('/dev/stdin', 'utf8').split('\n');
const n = parseInt(lines[0]);
const arr = lines[1].trim().split(/\s+/).map(Number).slice(0, n);

console.log(longestIncreasingStreak(arr));